include/cantera/oneD/StFlow.h

```
@@ -254,6 +255,87 @@ class StFlow : public Domain1D
+     * compute the mass flux at the right boundary.
+     *
+     * This method is based on the work of M. Nishioka, C.K. Law, and T. Takeno (1996) titled
+     * "A Flame-Controlling Continuation Method for Generating S-Curve Responses with
+     * Detailed Chemistry"
+     **/
+
+    //! Returns the temperature at the left control point
+    double leftControlPointTemperature() const {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            return m_tLeft;
+        } else {
+            return NAN;
+        }
+    }
+
+    //! Returns the z-coordinate of the left control point
+    double leftControlPointCoordinate() const {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            return m_zLeft;
+        } else {
+            return NAN;
+        }
+    }
+
+    //! Sets the temperature of the left control point
+    void setLeftControlPointTemperature(double temperature) {
+        if (m_twoPointControl && (m_zLeft != Undef)) {
+            m_tLeft = temperature;
+        }
+    }
+
+    //! Returns the temperature at the right control point
+    double rightControlPointTemperature() const {
+        if (m_twoPointControl && (m_zRight != Undef)) {
+            return m_tRight;
+        } else {
+            return NAN;
+        }
+    }
+
+    //! Returns the z-coordinate of the right control point
+    double rightControlPointCoordinate() const {
+        if (m_twoPointControl && (m_zRight != Undef)) {
+            return m_zRight;
+        } else {
+            return NAN;
+        }
+    }
+
+    //! Sets the temperature of the right control point
+    void setRightControlPointTemperature(double temperature) {
+        if (m_twoPointControl && (m_zRight != Undef)) {
+            m_tRight = temperature;
+        }
+    }
+
+    //! Sets the status of the two-point control
+    void enableTwoPointControl(bool twoPointControl) {
+        m_twoPointControl = twoPointControl;
+    }
```

This method should probably raise an exception if two point control doesn't make sense for the current flame. Or at least, there needs to be a check of some sort before calling `solve()`. Currently it seems like you mostly just end up with "Jacobian is singular" errors if things are incorrectly defined.

---

I used a check on the m_usesLambda for whether to throw an exception, as that is the marker for axisymmetric flows I believe.